Batfish crash when using non-existent node in Traceroute question

When traceroute is initiated from the Java console, Batfish handles the fact that the ingressNode specified doesn't exist:

```
batfish> set-testrig master_2017-10-19-12:00:00_da54ddb
Base testrig->env is now master_2017-10-19-12:00:00_da54ddb->env_default
batfish> get traceroute ingressNode="fin-app-02", dstIp=10.1.2.2
Status: FAILURE
Question: traceroute ingressNode=fin-app-02, dstIp=10.1.2.2
org.batfish.common.BatfishException: Failed to answer question
   at org.batfish.main.Batfish.answer(Batfish.java:566)
   at org.batfish.main.Batfish.run(Batfish.java:3926)
   at org.batfish.main.Driver$1.run(Driver.java:356)
Caused by: org.batfish.common.BatfishException: Cannot create flow with non-existent ingress node: 'fin-app-02'
   at org.batfish.question.TracerouteQuestionPlugin$TracerouteAnswerer.getFlows(TracerouteQuestionPlugin.java:129)
   at org.batfish.question.TracerouteQuestionPlugin$TracerouteAnswerer.answer(TracerouteQuestionPlugin.java:40)
   at org.batfish.main.Batfish.answer(Batfish.java:563)
   ... 2 more
Summary: 0 (non-assertion) results; 

```

However, when the call comes via the jsonpath question template, it crashes with a null pointer exception

```
samir-work:standard samir$ more answer.json 
{
  "answerElements" : [
    {
      "class" : "org.batfish.common.BatfishException$BatfishStackTrace",
      "answer" : [
        "org.batfish.common.BatfishException: Failed to answer question",
        "   at org.batfish.main.Batfish.answer(Batfish.java:566)",
        "   at org.batfish.main.Batfish.run(Batfish.java:3926)",
        "   at org.batfish.main.Driver$1.run(Driver.java:356)",
        "Caused by: java.lang.NullPointerException",
        "   at org.batfish.bdp.BdpEngine.collectFlowTraces(BdpEngine.java:151)",
        "   at org.batfish.bdp.BdpEngine.lambda$processFlows$44(BdpEngine.java:1566)",
        "   at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)",
        "   at java.util.TreeMap$KeySpliterator.forEachRemaining(TreeMap.java:2746)",
        "   at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)",
        "   at java.util.stream.ForEachOps$ForEachTask.compute(ForEachOps.java:291)",
        "   at java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:731)",
        "   at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)",
        "   at java.util.concurrent.ForkJoinTask.doInvoke(ForkJoinTask.java:401)",
        "   at java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:734)",
        "   at java.util.stream.ForEachOps$ForEachOp.evaluateParallel(ForEachOps.java:160)",
        "   at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateParallel(ForEachOps.java:174)",
        "   at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)",
        "   at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)",
        "   at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:583)",
        "   at org.batfish.bdp.BdpEngine.processFlows(BdpEngine.java:1523)",
        "   at org.batfish.bdp.BdpDataPlanePlugin.processFlows(BdpDataPlanePlugin.java:128)",
        "   at org.batfish.main.Batfish.processFlows(Batfish.java:3346)",
        "   at org.batfish.question.TracerouteQuestionPlugin$TracerouteAnswerer.answer(TracerouteQuestionPlugin.java:41)",
        "   at org.batfish.question.jsonpath.JsonPathQuestionPlugin$JsonPathAnswerer.answer(JsonPathQuestionPlugin.java:161)",
        "   at org.batfish.question.jsonpath.JsonPathQuestionPlugin$JsonPathAnswerer.answer(JsonPathQuestionPlugin.java:140)",
        "   at org.batfish.main.Batfish.answer(Batfish.java:563)",
        "   ... 2 more"
      ]
    }
  ],
  "question" : {
    "class" : "org.batfish.question.jsonpath.JsonPathQuestionPlugin$JsonPathQuestion",
    "debug" : false,
    "differential" : false,
    "innerQuestion" : {
      "class" : "org.batfish.question.TracerouteQuestionPlugin$TracerouteQuestion",
      "differential" : false,
      "dstIp" : "10.1.2.2",
      "dstPort" : 22,
      "dstProtocol" : "tcp",
      "ingressNode" : "fin-app-02",
      "srcIp" : "10.100.1.20"
    },

```

--- a/projects/question/src/main/java/org/batfish/question/TracerouteQuestionPlugin.java
+++ b/projects/question/src/main/java/org/batfish/question/TracerouteQuestionPlugin.java
@@ -3,6 +3,7 @@ package org.batfish.question;
 import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.google.auto.service.AutoService;
+import com.google.common.base.Strings;
 import java.util.Collections;
 import java.util.Map;
 import java.util.Set;
@@ -104,7 +105,7 @@ public class TracerouteQuestionPlugin extends QuestionPlugin {
             _batfish.popEnvironment();
           }
           String hostname = flowBuilder.getIngressNode();
-          Configuration node = (hostname == null) ? null : configurations.get(hostname);
+          Configuration node = Strings.isNullOrEmpty(hostname) ? null : configurations.get(hostname);
           if (node != null) {
             Set<Ip> ips =
                 new TreeSet<>(

--- a/projects/question/src/main/java/org/batfish/question/TracerouteQuestionPlugin.java
+++ b/projects/question/src/main/java/org/batfish/question/TracerouteQuestionPlugin.java
@@ -105,7 +105,8 @@ public class TracerouteQuestionPlugin extends QuestionPlugin {
             _batfish.popEnvironment();
           }
           String hostname = flowBuilder.getIngressNode();
-          Configuration node = Strings.isNullOrEmpty(hostname) ? null : configurations.get(hostname);
+          Configuration node =
+              Strings.isNullOrEmpty(hostname) ? null : configurations.get(hostname);
           if (node != null) {
             Set<Ip> ips =
                 new TreeSet<>(

